### PR TITLE
fix breeze Vulnerability

### PR DESCRIPTION
**Pull request by Le-Zheng**
dllib ->  breeze(compile, Apache 2.0) -> jtransforms(compile)
<html>
<body>

com.github.rwl:jtransforms@2.4.0 | GPL-3.0, LGPL-3.0, MPL-2.0
-- | --

</body>
</html>
[pom](https://repo1.maven.org/maven2/com/github/rwl/jtransforms/2.4.0/jtransforms-2.4.0.pom)

### CODE REVIEW

**Comment by Le-Zheng** [comment]
cannot remove http://10.112.231.51:18888/job/BigDL-PRVN-keras-scala-spark3.1/192/console:
```
[ERROR] /opt/work/jenkins/workspace/BigDL-PRVN-keras-scala-spark3.1/scala/dllib/src/test/scala/com/intel/analytics/bigdl/dllib/tensor/DenseTensorSpec.scala:24: object module is not a member of package spire.syntax
[ERROR] import spire.syntax.module
```

File: scala/dllib/pom.xml [hunk 1]
```diff
@@ -182,11 +182,12 @@
...
             <version>${project.version}</version>
-        </dependency>
-        <dependency>
+    	</dependency>
+	<dependency>
             <groupId>org.scalanlp</groupId>
             <artifactId>breeze_${scala.major.version}</artifactId>
-            <version>${breeze.version}</version>
+	    <version>${breeze.version}</version>
+	    <scope>${spark-scope}</scope>
```
Comment (glorysdj) [inline scala/dllib/pom.xml:190]: change to provided scope since breeze is only used in tests

Comment (Le-Zheng) [reply scala/dllib/pom.xml]: It seems also to be used in source code: 
[ERROR] /opt/work/jenkins/workspace/BigDL-PRVN-scalatest-optim-Spark-3.1/scala/dllib/src/main/scala/com/intel/analytics/bigdl/dllib/tensor/Tensor.scala:21: not found: object breeze
[ERROR] import breeze.linalg.{DenseMatrix => BrzDenseMatrix, DenseVector => BrzDenseVector}
http://10.112.231.51:18888/job/BigDL-PRVN-scalatest-optim-Spark-3.1/539/console

Comment (glorysdj) [reply scala/dllib/pom.xml]: try to exclude com.github.fommil.netlib:core in breeze for another try?

**Review by glorysdj — COMMENTED** [review]
shall we also remove other maven/plexus dependencies

**Comment by Le-Zheng** [comment]
> please run pr validation to check if exclude workds

jenkins pass http://10.112.231.51:18888/job/BigDL-PR-Dllib-Validation/535/
exclude `com.github.fommil.netlib:core`

**Review by glorysdj — APPROVED** [review]
LGTM

**Comment by glorysdj** [comment]
shall we also try exclude com.github.rwl:jtransforms@2.4.0?

**Comment by Le-Zheng** [comment]
> shall we also try exclude com.github.rwl:jtransforms@2.4.0?

Done. Jenkins: http://10.112.231.51:18888/job/BigDL-PR-Dllib-Validation/536/

**Pull request merged**